http_test.go

```
@@ -26,10 +30,89 @@ func (m *mockStore) Delete(key string) error {
 	return nil
 }
 
-func (m *mockStore) Close(key string) {
+func (m *mockStore) Close() {
 	return
 }
 
 func NewMockStore() *mockStore {
 	return &mockStore{make(map[string][]byte)}
 }
+
+
+
+func SetUp (t *testing.T) http.Handler {
+	store   := NewMockStore()
+	logger  := NewLogger("http_test")
+	handler := HttpHandler{store, &logger}
+	handle  := ParseKeyMiddleware(root, http.HandlerFunc(handler.Dispatch))
+
+	req, _  := http.NewRequest("POST", "http://localhost:8080/sessions/v1/foo", strings.NewReader("bar"))
```

At a minimum, we shouldn't ignore the error here; The error should be checked, and `t.Errorf()` invoked for failures.  However, since the value stored here exists solely to retrieve in `TestGet`, I wonder if it wouldn't be best to make the `POST` there instead.  That way, if it were to error, it would be encapsulated in that test.

Also, does that URI need to be fully qualified?  Referencing the previous comment, if you used a constant for the uri prefix, maybe you could reuse that here (ala `fmt.Sprintf("%sfoo", prefixUri)`, or something).

---

Great point about not ignoring the error. 

I wanted to test the post w/o the get but I see the confusion because I'm using the handler here. What if I just directly called the store and added it, something like `	store.Set("foo", []byte("bar"))`. That way it will be there for the get and I can try adding a new value with `POST`?

---

I can see wanting to avoid the `POST` in the `GET` test, because a failure there is an error with the tests, and not necessarily a `GET`-test failure, but I think this conundrum exists no matter what you do.  The way it is now, if it fails, it'll fail in `setUp`, which could theoretically happen for any of the tests (delete, post, etc).  If that happened, it would be surprising since the value `POST`-ed there isn't needed by them.

Now that I think about: A proper test of `POST` and `DELETE` should include a `GET` afterward to make sure the value is, and is not there, respectively, and this is also problematic.  TBH, this is why there was a single test originally...